packages/terra-application-utility/src/header/_HeaderUtilityMenuItem.jsx

```diff
@@ -0,0 +1,167 @@
+  /**
+   * The component associated with this menu item.
+   */
+  content: PropTypes.element,
+  /**
+   * The location of this menu item.
+   */
+  contentLocation: PropTypes.oneOf([Utils.LOCATIONS.BODY, Utils.LOCATIONS.FOOTER]),
+  /**
+   * Whether this item is selected.
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * Whether this item can be toggled.
+   */
+  isSelectable: PropTypes.bool,
+  /**
+   * Whether or not the menu item should display a disclosure indicator.
+   */
+  hasChevron: PropTypes.bool,
+  /**
+   * Function to trigger when this item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  title: '',
+  content: null,
+};
+
+class HeaderUtilityMenuItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.state = { isSelected: props.isSelected && props.isSelectable };
+  }
+
+  handleKeyDown(event, key) {
+    if (event.nativeEvent.keyCode === Utils.KEY_CODES.ENTER || event.nativeEvent.keyCode === Utils.KEY_CODES.SPACE || event.nativeEvent.keyCode === Utils.KEY_CODES.RIGHT_ARROW) {
+      this.props.onChange(event, key);
+    }
+  }
+
+  handleSelection(event, key) {
+    event.preventDefault();
+    if (this.props.isSelectable) {
+      const newIsSelected = !this.state.isSelected;
+      this.setState({ isSelected: newIsSelected });
+    }
+    this.props.onChange(event, key);
+  }
+
+  render() {
+    const {
+      itemKey,
+      title,
+      content,
+      contentLocation,
+      isSelected,
+      isSelectable,
+      hasChevron,
+      onChange,
+      ...customProps
+    } = this.props;
+
+    const bodyItemClassNames = cx(['body-item']);
+    const footerItemclassName = cx('footer-item');
+    const checkmarkClassName = cx(['checkmark', { selected: this.state.isSelected }]);
+    const chevronClassName = cx('chevron', { 'has-chevron': hasChevron });
+
+    // Footer items are buttons. Body items are list items.
+    // If content exists, render content. Else, render the title text.
+    /* eslint-disable jsx-a11y/no-static-element-interactions */
+    let item = null;
+    if (contentLocation === Utils.LOCATIONS.FOOTER) {
+      if (content) {
+        item = (
+          <button
+            {...customProps}
+            onClick={event => this.handleSelection(event, itemKey)}
+            onKeyDown={event => this.handleKeyDown(event, itemKey)}
+            className={footerItemclassName}
+            aria-label={title}
+          >
+            {content}
+          </button>
+        );
+      } else {
+        item = (
+          <button
+            {...customProps}
+            onClick={event => this.handleSelection(event, itemKey)}
+            onKeyDown={event => this.handleKeyDown(event, itemKey)}
+            className={footerItemclassName}
+            aria-label={title}
+          >
+            {title}
+          </button>
+        );
+      }
+    } else if (content) {
```

Comment:
Content should also be using the arrange, chevrons and selection shouldn't be part of the content.